src/contracts/core/AVSDirectoryStorage.sol

```diff
@@ -43,22 +47,46 @@ abstract contract AVSDirectoryStorage is IAVSDirectory {
     /// @notice Mapping: avs => operatorSetId => Whether or not an operator set is valid.
     mapping(address => mapping(uint32 => bool)) public isOperatorSet;
 
-    /// @notice Mapping: avs = operator => operatorSetId => Whether or not an operator is a member of an operator set.
-    mapping(address => mapping(address => mapping(uint32 => bool))) public isMember;
-
     /// @notice Mapping: avs => operatorSetId => Total operators within the given operator set.
     mapping(address => mapping(uint32 => uint256)) public operatorSetMemberCount;
 
     /// @notice Mapping: avs => Total amount of operator sets created for a given AVS.
     mapping(address => uint256) public avsOperatorSetCount;
 
+    /// @notice Mapping: operator => List of operator sets that operator is registered to.
+    /// @dev Each item is formatted as such: bytes32(abi.encodePacked(avs, uint96(operatorSetId)))
+    mapping(address => EnumerableSet.Bytes32Set) internal _operatorSetsMemberOf;
+
     /// @notice Total amount of operators sets created.
     uint256 public operatorSetCount;
 
     constructor(IDelegationManager _delegation) {
         delegation = _delegation;
     }
 
+    function memberFor(address operator, uint256 index) public view returns (OperatorSet memory) {
+        return _decodeOperatorSet(_operatorSetsMemberOf[operator].at(index));
+    }
+
+    function memberForCount(address operator) public view returns (uint256) {
+        return _operatorSetsMemberOf[operator].length();
+    }
+
+    function isMember(address operator, OperatorSet memory operatorSet) public view returns (bool) {
+        return _operatorSetsMemberOf[operator].contains(_encodeOperatorSet(operatorSet));
+    }
+
+    function _encodeOperatorSet(OperatorSet memory operatorSet) internal view returns (bytes32) {
+        return bytes32(abi.encodePacked(operatorSet.avs, uint96(operatorSet.operatorSetId)));
+    }
+
+    function _decodeOperatorSet(bytes32 encoded) internal view returns (OperatorSet memory) {
+        return OperatorSet({
```

Comment:
wdyt about using a custom type (OperatorSet) for bytes32

Reply:
Don't think it's needed here, will add better documentation though.